server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/http/DownloadRoutes.scala

```diff
@@ -0,0 +1,62 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.http
+
+import java.util.stream.Stream
+
+import org.apache.james.jmap.JMAPRoute.Action
+import org.apache.james.jmap.JMAPUrls.DOWNLOAD
+import org.apache.james.jmap._
+import org.slf4j.{Logger, LoggerFactory}
+
+object DownloadRoutes {
+  private val LOGGER = LoggerFactory.getLogger(DownloadRoutes.getClass)
+  private val BLOB_ID_PATH_PARAM = "blobId"
+  private val NAME_PATH_PARAM = "name"
+  private val DOWNLOAD_FROM_ID = String.format("%s/{%s}", DOWNLOAD, BLOB_ID_PATH_PARAM)
+  private val DOWNLOAD_FROM_ID_AND_NAME = String.format("%s/{%s}/{%s}", DOWNLOAD, BLOB_ID_PATH_PARAM, NAME_PATH_PARAM)
+}
+
+class DownloadRoutes extends JMAPRoutes {
```

Comment:
I'm pretty sure the download routes have a different structure in the new spec, not sure it makes sens to intoduce them now.